File: Umbrella/src/main/java/com/nerdery/umbrella/data/api/WeatherService.java
package com.nerdery.umbrella.data.api;

import com.nerdery.umbrella.BuildConfig;
import com.nerdery.umbrella.data.model.WeatherData;

import io.reactivex.Single;
import retrofit2.Call;
import retrofit2.adapter.rxjava2.Result;
import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;

/**
 * Retrofit interface for fetching weather data
 */
public interface WeatherService {

    /**
     * Get the forecast for a given zip code using {@link Call}
     */
    @GET("/api/" + BuildConfig.API_KEY + "/conditions/hourly/q/{zip}.json")
    Call<WeatherData> forecastForZipCallable(@Path("zip") String zipCode);

    /**
     * Get the forecast for a given zip code using {@link Observable}
     */
    @GET("/api/" + BuildConfig.API_KEY + "/conditions/hourly/q/{zip}.json")
    Single<Result<WeatherData>> forecastForZipObservable(@Path("zip") String zipCode);
}

File: src/main/java/com/nerdery/umbrella/data/api/WeatherService.kt
package com.nerdery.umbrella.data.api

import com.nerdery.umbrella.BuildConfig
import com.nerdery.umbrella.data.model.WeatherResponse
import com.nerdery.umbrella.data.model.TempUnit

import io.reactivex.Single
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface WeatherService {

    @GET("/forecast/${BuildConfig.API_KEY}/{latitude},{longitude}")
    fun getWeather(
        @Path("latitude") latitude: Double,
        @Path("longitude") longitude: Double,
        @Query("units") units: TempUnit
    ): Single<WeatherResponse>

    @GET("/forecast/${BuildConfig.API_KEY}/{latitude},{longitude}")
    fun getWeatherCall(
        @Path("latitude") latitude: Double,
        @Path("longitude") longitude: Double,
        @Query("units") units: TempUnit
    ): Call<WeatherResponse>
}